### DISCUSSION TITLE
A Patch to Increase Image and Rendition file name lengths to 235 characters

### DISCUSSION THREAD

**Discussion by ocuna** [wagtail/wagtail#8254] [Show and tell]
This is a simple experimental change to Wagtail 2.16.1 that enables a developer to increase the file name length of images and renditions from 100 up to 235 characters.  Patch includes tests.

After the patch is applied, adjust the file length by changing:
**wagtail.images.models.get_image_filename_maxlength(): to return any number between 100 and 235**

Please note, I found a solution to my problem that solves the problem this patch was designed to fix.  Here's the story: 
- Original file names of renditions can be severely truncated depending on the length of filter-spec attributes
    ex: "fill-810x525-c100 format-webp webpquality-60" will create a very long file name.
- Duplicate file names of renditions become common under long filter-specs as it truncates to file name to just a few characters before adding the filter spec.  Images with the same filter spec would overwrite (on my build).
- (this patch solution) Increases the length of file names reduces the possibility of over-written files

It turns out when I was testing my own patch it wasn't needed on a vanilla-install because the files were saved locally and would apply a unique code to the end of file names that already existed to keep them unique.  I had a setting in my cloud storages that would overwrite files instead of appending the unique code to keep them separate.  I assumed they were being truncated incorrectly and thus the patch was born.  So I decided to upload it here in case anyone wants to comment on it.
- (actual solution) Setting overwrite_files in storages.storages.backends.azure_storage from True to False

Maybe this will be useful to someone out there who wants longer file names by default.  It was a waste of time in the end for me to make it but I did learn a lot about wagtail and creating tests.

I would like any feedback as this is my first time writing a wagtail patch

[Wagtail-v2.16.1_MaxImageFilenameLength.patch.txt](https://github.com/wagtail/wagtail/files/8383078/Wagtail-v2.16.1_MaxImageFilenameLength.patch.txt)

**Comment by lb-** [comment]
Hey @ocuna thanks for sharing!

This sounds like a great potential blog post also, you can share your journey of how you solved your problem and how you learned Django/Wagtail testing and the inner workings of `AbstractImageRendition` along the way.

**Comment by muellert** [comment]
Having this configurable would be really great, preferably with a variable in settings. Eg. WAGTAIL_FILENAME_LENGTH or something.
